lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -1223,6 +1223,11 @@ def __init__(
         self.framework.observe(events.relation_changed, self._on_logging_relation_changed)
         self.framework.observe(events.relation_departed, self._on_logging_relation_departed)
 
+        for container_name in self._charm.unit.containers:
+            self.framework.observe(
+                self._charm.on[container_name].pebble_ready, self._on_logging_relation_changed
```

Comment:
I think we should track `relation_joined` instead, as that is fired up even on pod churn.

Reply:
Your points are valid. Just this morning I raised [similar concerns here](https://github.com/canonical/loki-k8s-operator/issues/112). 

However I am surprised by the statement that "`relation_joined` is fired up even on pod churn" given the [official docstring](https://ops.readthedocs.io/en/latest/#ops.charm.RelationJoinedEvent).

Reply:
> The event fires only when that remote unit is first observed by the unit.

When a pod churns, IIRC it counts as a new unit. Right @jameinel?

Reply:
That might very well be the case, but it is not what was communicated when we created the Juju state machine diagram.

At that point, if I'm not completely mistaken, it was stated that a pod churn won't count as a new unit, as for all juju cares, it's the same (ie `charm/0` is still `charm/0`).

Reply:
Iirc, on upgrade we do not see any relation events.